### ISSUE TITLE
[BUG] changes to tool configuration cause compilation results to be invalidated

### ISSUE THREAD

**Issue by mattgodbolt** [mattgodbolt/compiler-explorer#1482]
Editing options in the tools attached to a compiler causes the compilation to be redone (and cached results not reused).

**To Reproduce**
* Go to https://gcc.godbolt.org/z/BO5N7g
* Edit the `llvm-mca` tool options (e.g. add some whitespace)
* Observe the compilation takes a long while again (even though in theory it should be cached).

**Expected behavior**
Compilation results should be reused, and only the tool re-executed.

**Comment by partouf** [comment]
It depends on the tool what kind of input is taken (asm file, asm stdin, obj file, exe file, etc), so we'd need to facilitate new types of caching that support all these ways.

**Comment by mattgodbolt** [comment]
Right. Good point @partouf , this is more like an enhancement for caching to be more fine-grained.
